### DISCUSSION TITLE
Giving sonic the bounce

### DISCUSSION THREAD

**Discussion by zacnmac** [Techokami/SonicWorldsNext#90] [Q&A]
I want to give Sonic the ability to bounce like you can with the bubble shield at any time. This move will replace the dropdash. How would I do this?

**Comment by DimensionWarped** [comment]
Start by checking Air.gd (Scripts/Player/States/Air.gd) -- there's a big match statement for the parent.character which checks if the player executing the air-state is Sonic, Tails, Knuckles, or Amy (just a heads up, this is a likely target for refactoring later since it'd generally be better for players to have their own set of callbacks for these that gets configured in the player object). In the Sonic object there are also matches for the parent.shield. In here you control the behavior around various shields.

Worth noting though that 'replace the dropdash' is weird phrasing because the drop dash is set up *after* a double jump action takes place while the various shield actions are double jump actions. The drop dash code also lives in this file (search "# Drop dash for sonic / amy" to get there quickly) and I suppose you could technically copy aspects of the bounce attack to it, but I'm not sure how much sense it would actually make (IE shield move in Sonic's case) to going into drop dash. Feel free to experiment though.

**Reply by zacnmac** [reply]
Is there maybe a way to completely remove Sonic's shield move? I may or may not be making a game on a certain armadillo as opposed to Sonic..

**Reply by DimensionWarped** [reply]
Yes, of course. You'd just remove the match statement around shields in the same section of code that I mentioned before.
